### ISSUE TITLE
Document that Map DOM element must be empty

### ISSUE THREAD

**Issue by Tr1po** [mapbox/mapbox-gl-js#3020]
Hi,
I am using mapbox-gl  to connect cities around the world with line. My main problem is that when I add Marker or Popup it is not on right position in scale I need it, because when I zoom it, it is getting nearer and nearer to its right position. However, I need to display it on right position in "whole-world" scale, because I want to show whole world with no possibility for zoom. Here is my code for Popup CityName.

`componentDidMount() {
        const { airport, map } = this.props;
        new mapboxgl.Popup({ closeOnClick: false, closeButton: false, anchor: 'bottom-left' })
            .setLngLat([airport.longitude, airport.latitude])
            .setDOMContent(findDOMNode(this))
            .addTo(map);
    },

```
render() {
    return (
        <div className={styles.cityName}>
            {this.props.airport.name}
        </div>
    );
},`
```

![screen shot 2016-08-17 at 14 12 48](https://cloud.githubusercontent.com/assets/19331847/17735853/dccd4b7a-6484-11e6-8e9e-856d60737ce8.png)
I am sure that locations are right. I wanted to use for city markers Marker but it had the same problem, so I did it through Layer. Nevertheless, it is a little more complicated and not so easy to interact with, it would help me a lot to use markers and popups. Amazing thing is when I change popup element (cityName) size to 2x2px it gets much nearer, I would say to position as it should be.
![screen shot 2016-08-17 at 14 13 29](https://cloud.githubusercontent.com/assets/19331847/17735949/74b09b72-6485-11e6-8a28-ac5dcd6fff5b.png)

It would be great if somebody can help me. Thanks in advance!!

**Comment by mourner** [comment]
Can you set up a minimal test case (preferably without React) on JSFiddle that reproduces this issue?

**Comment by Tr1po** [comment]
Here is JSFiddle clarification https://jsfiddle.net/s7L3j7r7/1/ (cities are Chicago, London, Bangkok), it seems like in Map DOM element can't be any DOM element, otherwise it conflicts with markers and popups positions. I think it would be great if it was written in documentation.

**Comment by mourner** [comment]
Could you please describe what's going on in that JSFiddle? I see a static map with 3 popups open, one of them with a blue block. What do you expect to see?

**Comment by Tr1po** [comment]
It doesn't have any meaning. All it should do and does is to show that popups are not on correct positions on map.

**Comment by jfirebaugh** [comment]
The map DOM element should not have children as in the JSFiddle; Mapbox GL expects to control the contents of that element itself. We can add a note about this to the documentation.

https://jsfiddle.net/urve42n5/ works as expected.

**Comment by jfirebaugh** [comment]
Fixed in cf2900587c99381cfa37fd0e03e0981e50571253.
